src/mjolnir/graphtilebuilder.cc

```diff
@@ -224,16 +218,27 @@ GraphTileBuilder::GraphTileBuilder(const std::string& tile_dir,
 
   // Text list
   for (auto ni = name_info.begin(); ni != name_info.end(); ++ni) {
-    // TODO: this is a bug. the last entry will include the padding with it, for a non tagged value this probably isnt
-    //  a big deal, but for tagged values this means the value can be compromised
     // compute the width of the entry by looking at the next offset or the end if its the last one
     auto next = std::next(ni);
-    auto width = next != name_info.end() ? (next->name_offset_ - ni->name_offset_)
-                                         : (textlist_size_ - ni->name_offset_);
+
+    size_t width;
+
+    if (next != name_info.end()) {
+      // Non-last entry: use the next entry's offset
+      width = next->name_offset_ - ni->name_offset_;
+    } else {
+      // Last entry: for tagged values, use TaggedValueSize to avoid including padding bytes
+      // that were added for alignment. For non-tagged values just read to the end.
+      if (ni->tagged_) {
+        width = EdgeInfo::TaggedValueSize(textlist_ + ni->name_offset_);
+      } else {
+        width = textlist_size_ - ni->name_offset_;
+      }
```

Comment:
cant we reduce the nesting a bit:

```suggestion
    // Last entry: for tagged values, use TaggedValueSize to avoid including padding bytes
    // that were added for alignment. For non-tagged values just read to the end.
    } else if (ni->tagged_) {
      width = EdgeInfo::TaggedValueSize(textlist_ + ni->name_offset_);
    // Last entry is just text so use the text list ptr as its the next thing in the tile
    } else {
      width = textlist_size_ - ni->name_offset_;
```